src/Adapter/Cart/CommandHandler/SendCartToCustomerHandler.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Cart\CommandHandler;
+
+use Cart;
+use Context;
+use Customer;
+use Language;
+use Mail;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Command\SendCartToCustomerCommand;
+use PrestaShop\PrestaShop\Core\Domain\Cart\CommandHandler\SendCartToCustomerHanlderInterface;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Exception\CartException;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Exception\CartNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Cart\ValueObject\CartId;
+use Validate;
+
+/**
+ * @internal
+ */
+final class SendCartToCustomerHandler implements SendCartToCustomerHanlderInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(SendCartToCustomerCommand $command)
+    {
+        $cart = $this->getCart($command->getCartId());
+        $customer = $this->getCustomer($cart->id_customer);
+
+        $mailVars = [
+            '{order_link}' => $this->generateCheckoutUrl($cart),
+            '{firstname}' => $customer->firstname,
+            '{lastname}' => $customer->lastname,
+        ];
+
+        $cartLanguage = new Language((int) $cart->id_lang);
+
+        $emailWasSent = Mail::send(
```

Comment:
The idea is nice, but not sure if this is a right time. :/

Reply:
OK then 👍but let's create an Issue for it. I would like to avoid having mails being sent from random locations in the migrated code. Also having a unique MailSender would allow us to put an Interface on it, first implement it with an Adapter, but could be replaced later by any modern php mail client. Easier also to mock, disable, put options.

Reply:
Should I do that or you will? :)

Reply:
When the PR is merged, so that we can refer to the right line of code